src/main/java/com/vention/stock_market_share/service/EmailService.java

```diff
@@ -0,0 +1,61 @@
+package com.vention.stock_market_share.service;
+
+import jakarta.mail.internet.MimeMessage;
+import jakarta.mail.util.ByteArrayDataSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMailMessage;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+public class EmailService {
+    private final JavaMailSender mailSender;
+
+    public EmailService(JavaMailSender mailSender) {
+        this.mailSender = mailSender;
+    }
+
+    @Value("${spring.mail.username}")
+    private String fromMail;
+
+    public String sendMailWithLink(String to, String link) {
+        try {
+            SimpleMailMessage message = new SimpleMailMessage();
+            message.setFrom(fromMail);
+            message.setTo(to);
+            message.setSubject("Registration Confirmation");
+            message.setText("Click the following link to confirm your registration: " + link);
+            mailSender.send(message);
+            return "mail sent";
+        } catch (Exception e) {
+            throw new RuntimeException();
+        }
+    }
+
+    public String sendMail(MultipartFile[] file, String to, String[] cc, String subject, String body) {
+        try {
+            MimeMessage mimeMessage = mailSender.createMimeMessage();
+            MimeMessageHelper mimeMessageHelper = new MimeMessageHelper(mimeMessage, true);
+            mimeMessageHelper.setFrom(fromMail);
+            mimeMessageHelper.setTo(to);
+            mimeMessageHelper.setCc(cc);
+            mimeMessageHelper.setSubject(subject);
+            mimeMessageHelper.setText(body);
+            for (int i = 0; i < file.length; i++) {
+                mimeMessageHelper.addAttachment(
+                        file[i].getOriginalFilename(),
+                        new ByteArrayResource(file[i].getBytes())
+                );
+            }
+            mailSender.send(mimeMessage);
+            return "mail sent";
+
+        } catch (Exception e) {
+            throw new RuntimeException();
```

Comment:
do not throw exception in catch block. Just log it here.

Reply:
Done